sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -278,27 +274,48 @@ public boolean isRetryAfter(final @NotNull String itemType) {
   @Override
   public @NotNull TransportResult send(final @NotNull SentryEnvelope envelope) throws IOException {
     final HttpURLConnection connection = createConnection(true);
-
-    int responseCode = -1;
+    TransportResult result;
 
     try (final OutputStream outputStream = connection.getOutputStream();
         final GZIPOutputStream gzip = new GZIPOutputStream(outputStream);
         final Writer writer = new BufferedWriter(new OutputStreamWriter(gzip, UTF_8))) {
       serializer.serialize(envelope, writer);
-
-      logger.log(DEBUG, "Envelope sent successfully.");
```

Comment:
this should not have been here in the first place, as we are using `try-with-resources`, the resources are only available and freed to check its response code in the finally block, as the `try-with-resources` finishes every stream at the end, so the proper way to verify and log would be in the final block.